superset/connectors/sqla/models.py

```diff
@@ -2089,21 +1981,24 @@ def after_update(  # pylint: disable=too-many-branches, too-many-locals, too-man
         # set permissions
         security_manager.set_perm(mapper, connection, target)
 
-        dataset = (
-            session.query(NewDataset).filter_by(sqlatable_id=target.id).one_or_none()
-        )
-        if not dataset:
-            return
-
         # get DB-specific conditional quoter for expressions that point to columns or
         # table names
         database = (
             target.database
             or session.query(Database).filter_by(id=target.database_id).one()
         )
+
         engine = database.get_sqla_engine(schema=target.schema)
         conditional_quote = engine.dialect.identifier_preparer.quote
 
+        dataset = (
+            session.query(NewDataset).filter_by(sqlatable_id=target.id).one_or_none()
+        )
+
+        if not dataset:
+            dataset = SqlaTable.write_shadow_dataset(target, database, session)
```

Comment:
Small nit for consistency:

```suggestion
            SqlaTable.write_shadow_dataset(target, database, session)
```